### ISSUE TITLE
one colorscheme is not applied well in SpaceVim

### ISSUE THREAD

**Issue by jetm** [joshdick/onedark.vim#99]
- [X] I have carefully read and followed the [Installation section of the README](https://github.com/joshdick/onedark.vim#installation)
- [X] I have searched through the [closed issues](https://github.com/joshdick/onedark.vim/issues?q=is%3Aissue+is%3Aclosed) to see if there is an existing solution for my issue

### Terminal Emulator Info

konsole and xfce-terminal

### Output From `vim --version`

For further details: [Spacevim's Issue 507.](https://github.com/SpaceVim/SpaceVim/issues/507)

**Comment by joshdick** [comment]
I don't use SpaceVim, but can you elaborate on what "not applied well" means? Does the colorscheme simply not work at all like you mentioned in SpaceVim/SpaceVim#507, or is the theme displaying, but with incorrect colors?

Your report looks somewhat like SpaceVim/SpaceVim#63, do you have `:syntax on` in your configuration?

**Comment by jetm** [comment]
@joshdick: It does apply, but not all. You can see it in the screenshot. The rare part for me is when I run `colorscheme` shows 'default'. There is a Vim colorscheme called 'default'.  If I execute 'colorscheme onedark' the colors is fixed. Also, in the screenshot I am using tmux, but I get the same result without tmux.

![screenshot_2017-05-09_10-30-35](https://cloud.githubusercontent.com/assets/474838/25861631/9232c3e4-34a2-11e7-8809-8a4e205a2cf4.png)

**Comment by jetm** [comment]
@joshdick: you workaround fixed this issue. Thanks!

**Comment by joshdick** [comment]
Fixed as of 22c8bb6. Copied from [my comment](https://github.com/SpaceVim/SpaceVim/issues/507#issuecomment-300328082) in SpaceVim/SpaceVim#507 for future reference:

---

I spent quite a while debugging this.

It seems to be caused by an odd combination of SpaceVim exacerbating a Vim bug.

Here's some information about the Vim bug:

- https://github.com/SpaceVim/SpaceVim/blob/7e9742a171a6b551b0cc35e5dc180089b19dc444/config/general.vim#L10

- https://www.reddit.com/r/vim/comments/22krs1/using_set_backgrounddark_in_vim_resets_the/

And here's the [relevant code in SpaceVim](https://github.com/SpaceVim/SpaceVim/blob/7e9742a171a6b551b0cc35e5dc180089b19dc444/config/general.vim#L10):

```vim
if count(g:spacevim_plugin_groups, 'colorscheme') && g:spacevim_colorscheme !=# '' "{{{
    try
        exec 'colorscheme ' . g:spacevim_colorscheme
        exec 'set background=' . g:spacevim_colorscheme_bg
    catch
        exec 'colorscheme '. g:spacevim_colorscheme_default
    endtry
else
    exec 'colorscheme '. g:spacevim_colorscheme_default
endif
```

Commenting out the `exec 'set background=' . g:spacevim_colorscheme_bg` makes onedark.vim load with proper/configured colors as expected.

Here's what appears to be happening:

1. `exec 'colorscheme ' . g:spacevim_colorscheme` is run, which for onedark.vim, should be the equivalent of `colorscheme onedark`.

2. At this point, running `:set background` appears to yield `light`, which is part of the Vim bug; `background` should be `dark` since `onedark.vim` is explicitly setting it that way.

3.  `exec 'set background=' . g:spacevim_colorscheme_bg` is run, which should be the equivalent of `set background=dark`. **This causes all colors to be reset since the background color has changed from light to dark**, which is the other part of the Vim bug, and which is causing the behavior you're seeing.

One possible workaround is to move `set background=dark` to the bottom of the affected colorschemes, but I'm not sure how to test that locally because I can't figure out how to get around SpaceVim's (apparently many) layers of caching installed plugins.

`neodark` includes `set background=dark` at the bottom, which is probably why it seems to be unaffected by this bug.
